blog/02-01-2023-flashloan-aave-celo/flashloan aave celo.md

```diff
@@ -0,0 +1,254 @@
+
+* [FlashLoanReceiverBase.sol](https://gist.github.com/johnfawole/741aff645a7e375f8fbccb0fab5ee910)
+* [Interfaces.sol](https://gist.github.com/johnfawole/c2844bd1450607ead2969b8f83e340a3)
+* [Libraries.sol](https://gist.github.com/johnfawole/353940d436643123603a8375703646ec)
+* [Ownable.sol](https://gist.github.com/johnfawole/8e37c45d86315744339b952a2846ff84)
+
+![contracts](contracts.jpeg)
+
+The MyFlashLoan contract inherits from the FlashLoanReceiverBase and Ownable contract.
+
+On line 20, we are passing in a constructor argument of an address to the inherited FlashLoanReceiverBase contract which determines which LendingPoolAddressProvider to use on the network the contract is deployed. 
+
+We can find the _addressProvider from [the documentation](https://docs.aave.com/developers/v/2.0/deployed-contracts/deployed-contracts) as we'll be passing in the relevant LendingPoolAddressProvider value at deployment.
+
+## Step 2: Implementing your Flash Loan function
+Next, add the function that will be called to initiate the Flash Loan process. Let's call the function myFlashLoanCall() and this function will take the address of the token you want to flashloan as an argument. 
+
+We will limit access to it to the owner (i.e. you), this will help guard against griefing attacks. You will request a Flash Loan of 1 USDC (1* 10^6 as USDC has 6 decimal places). We don't need data for the flashloan, so we are passing an empty string. 
+
+This function then calls the flashloan() function on the lending pool contract for you to receive the tokens.
+```
+    function myFlashLoanCall(address _tokenAddress) public onlyOwner {
+        address receiverAddress = address(this);
+
+        address[] memory assets = new address[](1);
+        assets[0] = address(_tokenAddress); //goerli USDC 0x9FD21bE27A2B059a288229361E2fA632D8D2d074
+
+        uint256[] memory amounts = new uint256[](1);
+        amounts[0] = 1*10**6;
+
+        // 0 = no debt, 1 = stable, 2 = variable
+        uint256[] memory modes = new uint256[](1);
+        modes[0] = 0;
+
+        address onBehalfOf = address(this);
+        bytes memory params = "";
+        uint16 referralCode = 0;
+
+        LENDING_POOL.flashLoan(
+            receiverAddress,
+            assets,
+            amounts,
+            modes,
+            onBehalfOf,
+            params,
+            referralCode
+        );
+    }
+}
+```
+
+## Step 3: Executing your Flash Loan operation
+If we have requested a valid amount from a valid reserve, then the LendingPool contract will call the executeOperation() function in our contract. This is where we utilize the funds received from the lending pool. It checks if we received the appropriate loan amount or will throw an error message.
+As shown below, we can include our executeOperation() function:
+```
```

Comment:
add a space before the code snippet and add the intended code language like this 
" ```solidity
// code here
``` "